### DISCUSSION TITLE
Nix package manager as an alternative to brew

### DISCUSSION THREAD

**Discussion by marc0der** [blue-build/.github#43] [General]
Hi folks :wave: 

I want to hear what the feasibility would be of enabling Nix as an alternative to Brew. I've been using Nix package manager and hom-manager on top of my Fedora Atomic Sway desktop for some time now, and have had great success with this approach. It does both package and dotfiles management sweetly, and it seems like a great fit for Bluebuild. I think all downstream projects could benefit from this alternative too (I wouldn't mind having Nix on my steam deck with Bazzite, for instance). Would this be something that would be considered?

I'd be happy to get my hands dirty too, and if everyone thinks it's a good idea, I can contribute this functionality to the Bluebuild project.

Curious to hear your thoughts and ideas about this!

**Comment by fiftydinar** [comment]
One issue I remember with `nix` is conflicts with SeLinux due to its unique file structure.
Universal Blue removed the nix from ujust for this reason:
https://github.com/ublue-os/config/pull/226

But I didn't use it, nor I know much about it, so I'll let someone else to chime in with more details.

**Comment by xynydev** [comment]
I used it, and I didn't really encounter much problems. There was one with the Steam Flatpak, though, preventing it from lauching completely (not sure if that's still an issue tbh). I think it was about SELinux too.

So yeah, it's a useful and good tool, and the determinate systems installer gets it mostly working, but there are still likely some prevailing issues.

@RoyalOughtness might know more about the issues regarding SELinux.

---

As for a contribution; a module that installs Nix (if possible, with determinate systems) such that it is integrated with the system at build time would be accepted. If that's something that's found to be possible, that's cool regardless of possible issues and we'll js have to point those out in the documentation.

If that's not possible, however, I'm not sure it makes sense to make a module that just sets the installer to run on first boot automatically. A power user using Nix would probably be capable, willing, and happy to run the install themselves.
